### PR TITLE
Remeber reply state

### PR DESCRIPTION
**Pull request by cameronaaron**

### CODE REVIEW

**Review by cshariq — APPROVED** [review]
Compiled with this commit, works fine.

**Comment by cameronaaron** [comment]
> Compiled with this commit, works fine.

This pull is a work in progress. Are you having any bugs or noticing any benefits ??

File: lib/services/ui/chat/conversation_view_controller.dart [hunk 1]
```diff
@@ -232,4 +233,27 @@ class ConversationViewController extends StatefulController with GetSingleTicker
     cm.setAllInactiveSync();
     Get.delete<ConversationViewController>(tag: tag);
   }
-}
\ No newline at end of file
+
+  Future<void> saveReplyToMessageState() async {
+    final prefs = await SharedPreferences.getInstance();
```
Comment (zlshames) [inline lib/services/ui/chat/conversation_view_controller.dart]: Pretty sure we already have an instance of shared preferences in the settings service. Use `ss.prefs` to access it.

File: lib/services/ui/chat/conversation_view_controller.dart [hunk 2]
```diff
@@ -232,4 +233,27 @@ class ConversationViewController extends StatefulController with GetSingleTicker
...
+      await prefs.setString('replyToMessage_${chat.guid}', replyToMessage!.item1.guid!);
+      await prefs.setInt('replyToMessagePart_${chat.guid}', replyToMessage!.item2);
+    } else {
+      await prefs.remove('replyToMessage_${chat.guid}');
+      await prefs.remove('replyToMessagePart_${chat.guid}');
+    }
+  }
+
+  Future<void> loadReplyToMessageState() async {
+    final prefs = await SharedPreferences.getInstance();
```
Comment (zlshames) [inline lib/services/ui/chat/conversation_view_controller.dart]: Same comment about the shared prefs instance here

**Review by zlshames — CHANGES_REQUESTED** [review]

File: lib/services/ui/chat/conversation_view_controller.dart (hunk 2 above)
Comment (cameronaaron) [reply lib/services/ui/chat/conversation_view_controller.dart]: Dope Will do!

**Review by zlshames — APPROVED** [review]

**Pull request merged**